### PR TITLE
Fix Quantization and ONNX export for stand alone top/classifier layer

### PR DESCRIPTION
**Pull request by natuan**
This PR fixes the quantization and ONNX export for models where the classifier layer does not have submodules. In that case, there're 3 issues: (1) the add_quant_dequant func when applied to this layer (assuming it's specified in the quantization recipe) leaves it unchanged, instead of being wrapped by quant and dequant constructs; (2) the onnx export disconnects the model output from the output of the classifier's GEMM; (3) the name of the model output is set to dummy strings "output_xyz".

An example of the correct output of the classifier:
after_fix

### CODE REVIEW

**Review by bfineran — DISMISSED** [review]
LGTM, thanks for these fixes @natuan. small nit, can we change the name of `Mod2Quant` to something without the `2`?

File: src/sparseml/pytorch/optim/modifier_quantization.py [hunk 1]
```diff
@@ -49,6 +50,8 @@
     "QuantizationModifier",
 ]
 
+Mod2Quant = collections.namedtuple("Mod2Quant", ["name", "module"])
```
Comment (markurtz) [inline src/sparseml/pytorch/optim/modifier_quantization.py:53]: @natuan for consistency, please change over to NamedTuple from typings. Additionally, we need to add types for the attributes in this collection. Finally, if this is meant to be a private, internal variable please mark it as such by prefacing with an underscore: https://github.com/neuralmagic/sparseml/blob/a574de1dbac844080e986783ef7c88858c9a3a29/src/sparseml/onnx/utils/helpers.py#L226

Also, I agree with Ben that we want to stay away from putting numbers in our variable names. We've kept names verbose and clean for our current implementations and don't want to move away from that

File: src/sparseml/pytorch/optim/modifier_quantization.py [hunk 2]
```diff
@@ -364,15 +367,17 @@ def _enable_module_qat(self, module: Module):
 
         # prepare each module / submodule for quantization
         qconfig = get_qat_qconfig()
-        for quant_module in self._modules_to_quantize:
+        for mod2quant in self._modules_to_quantize:
```
Comment (markurtz) [inline src/sparseml/pytorch/optim/modifier_quantization.py:370]: please change out this variable name as well

**Review by markurtz — CHANGES_REQUESTED** [review]

**Comment by natuan** [comment]
> LGTM, thanks for these fixes @natuan. small nit, can we change the name of `Mod2Quant` to something without the `2`?

Changed it to ModuleToQuantize

File: src/sparseml/pytorch/optim/modifier_quantization.py (hunk 1 above)
Comment (natuan) [reply src/sparseml/pytorch/optim/modifier_quantization.py]: Changed to _ModuleToQuantize

File: src/sparseml/pytorch/optim/modifier_quantization.py (hunk 2 above)
Comment (natuan) [reply src/sparseml/pytorch/optim/modifier_quantization.py]: Fixed

**Review by markurtz — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**